lib/redlock/client.rb

```diff
@@ -102,6 +102,21 @@ def lock!(resource, *args)
       end
     end
 
+    # Gets remaining ttl of a resource
+    # Params:
+    # +lock_info+:: the lock that has been acquired when you locked the resource
+    def get_remaining_ttl(lock_info)
+      try_get_remaining_ttl(lock_info[:resource], lock_info[:value])
+    end
+
+    # Checks if a resource is locked
+    # Params:
+    # +lock_info+:: the lock that has been acquired when you locked the resource
+    def locked?(lock_info)
+      ttl = get_remaining_ttl(lock_info)
+      !(ttl.nil? || ttl.zero?)
+    end
```

Comment:
Now that I see it in code: Do you think adding this method is "safe" for users? I mean, you're excluding `ttl == 0`, but what if `ttl == 1`? `locked?` will return `true` and some innocent user of the library will happily continue with their critical code section, even though the lock will be released basically the next millisecond.

In other words: How about we don't add this `locked?` function at all, forcing users to think about their "minimum" remaining ttl to let their jobs continue?

Reply:
@maltoe that's a great observation. I'm thinking about my use case, where sometimes, for non-critical reasons like displaying an indicator back to the user, I do indeed just want to check if a resource is locked, and am not very worried if it will unlock immediately after. Here `locked?` is still useful. Perhaps this is something that could be addressed in the documentation instead to make it clear that `locked?` could behave this way, and if you need safety, you should use `get_remaining_ttl`. But we could also delegate this responsibility to the caller. What do you think?